### ISSUE TITLE
wrong value for sin cos

### ISSUE THREAD

**Issue by Shannon-John** [vnmakarov/mir#257]
Dear author:

I test c2m with c code incuding some trigonometric functions, but the result is not correct. Following is my code.

int
main()
{
	double s[16];
	double d[16];
	int i;
	
	for(i = 0; i < 16; i++)
		s[i] = 7.93;
	
	printf("done! %.5f\n", s[0]);
	
	d[0] = sin(s[0]);
	d[1] = cos(s[0]);
	printf("done! %.5f\n", d[0]);
	printf("done! %.5f\n", d[1]);
	
	return 0;
}

c2m test.bmir -eg
done! 7.93000
done! 1019.00000
done! 1019.00000

**Comment by vnmakarov** [comment]
Thank you for trying MIR.  I've reproduced the wrong result you are reporting (although it is a bit different).

The problem is that you missed add `#include <math.h>`.  So sin and cos are treated as functions returning integer through general purpose register although sin and cos return floating point numbers through xmm regs.  Basically `c2m` uses some garbage values in this case.
 
GCC is smart enough to understand that sin/cos from `libm` and uses the right prototypes for them.

The bottom line you should add `#include <math.h>` and the problem will be gone.
